src/librustdoc/doctest/make.rs

```diff
@@ -351,8 +347,6 @@ fn parse_source(source: &str, crate_name: &Option<&str>) -> Result<ParseSourceIn
         }
         match item.kind {
             ast::ItemKind::Fn(ref fn_item) if !info.has_main_fn => {
-                // We only push if it's the top item because otherwise, we would duplicate
-                // its content since the top-level item was already added.
```

Comment:
Outdated since #138104 when I suggested to remove the recursion in https://github.com/rust-lang/rust/pull/138104#discussion_r2015246273.